### PR TITLE
fix(bulk_update): skip empty trailing `emit_buffer` call

### PR DESCRIPTION
**Pull request by gkorland**
## Summary
After the row loop in `process_update_csv`, `emit_buffer` was always called even when `rows_strs` was empty (all rows already flushed in-loop, or no data rows present). This sent a useless `CYPHER rows=[] …` to the server.

## Changes
- `falkordb_bulk_loader/bulk_update.py`: only build and emit the trailing buffer when `rows_strs` is non-empty.

## Testing
`uv run flake8 falkordb_bulk_loader` clean.

## Memory / Performance Impact
Removes one unnecessary network round-trip per update run when the in-loop flush already drained `rows_strs`.

## Related Issues
From the comprehensive code-review report (BUG-15).

## Summary by CodeRabbit

* **Bug Fixes**
  * Resolved an issue where the bulk loader would emit unnecessary empty bulk-update commands when processing CSV files with only headers or no data rows.

* **Tests**
  * Added test coverage for edge cases in CSV processing to ensure proper handling of files with headers-only and empty content.

### CODE REVIEW

File: falkordb_bulk_loader/bulk_update.py [hunk 1]
```diff
@@ -127,8 +127,9 @@ def process_update_csv(self):
                     rows_strs.append(next_line)
                     self.buffer_size += added_size
             # Concatenate all rows into a valid parameter set
-            buf = "".join(["CYPHER rows=[", ",".join(rows_strs), "]"])
-            self.emit_buffer(buf)
+            if rows_strs:
+                buf = "".join(["CYPHER rows=[", ",".join(rows_strs), "]"])
+                self.emit_buffer(buf)
```
[the remark this answers is not in the document]
Comment (gkorland) [reply falkordb_bulk_loader/bulk_update.py:132]: @copilot apply changes based on [this feedback](https://github.com/FalkorDB/falkordb-bulk-loader/pull/59#discussion_r3143852762)

**Comment by gkorland** [comment]
## Copilot Code Review

✅ **No significant issues found.**

**Comment by gkorland** [comment]
## Agent Review Summary

### Phase 1 — PR Understanding
**PR**: `fix(bulk_update): skip empty trailing emit_buffer call`

**Intent**: Guard the post-loop `emit_buffer` call in `process_update_csv` so it only executes when `rows_strs` is non-empty, preventing a useless `CYPHER rows=[]` round-trip to the server when the CSV produces no data rows.

**Change**: Single 2→3 line diff in `bulk_update.py` — added `if rows_strs:` guard. Simple and correct.

### Phase 2 — Review Comment Classification
| Comment | Author | Classification | Action |
|---|---|---|---|
| "no tests for empty-input behavior" | copilot-pull-request-reviewer | ✅ Actionable | Already addressed in commit 250af4f |
| `@copilot apply changes based on this feedback` | gkorland | ✅ Actionable | Already addressed |
| "Added TestBulkUpdateEmptyCSV…" | copilot-pull-request-reviewer | �� Confirmation | No action needed |

All actionable review comments have been addressed prior to this review pass.

### Phase 3 — Code Review Assessment
The fix is correct. The original code always called `emit_buffer` after the loop regardless of whether any rows were accumulated (e.g. header-only CSV or empty file), sending a semantically invalid `CYPHER rows=[]` query. The guard is minimal and surgically correct.

**Tests added** (`TestBulkUpdateEmptyCSV`):
- `test_header_only_csv_does_not_emit` — header-only CSV → `emit_buffer` never called ✅
- `test_empty_csv_no_header_does_not_emit` — empty file with `--no-header` → `emit_buffer` never called ✅

### Phase 4 — Test Coverage
Both new tests pass locally. Coverage report (Codecov): 89.54% — all modified lines covered.

### Phase 5 — Rebase & CI
Branch was **4 commits behind** `master` (dependency bumps + two unrelated fixes). Rebased with `git rebase origin/master` and force-pushed with `--force-with-lease`.

**CI Result**: ✅ All 9 checks green
- `CI/Code linting` ✅
- `CI/Test with Python 3.10` ✅
- `CI/Test with Python 3.11` ✅
- `CodeQL (actions + python)` ✅
- `codecov/patch` and `codecov/project` ✅

### Verdict
✅ PR is **ready to merge**. The fix is correct, well-tested, and CI is fully green on top of the latest master.

**Review by shahar-biron — APPROVED** [review]
LGTM after code review.

Reviewed the diff for correctness and scope. Will merge once the branch is up to date with master and required CI checks are green.
